src/test/java/org/jitsi/meet/test/DesktopSharingTest.java

```diff
@@ -37,128 +31,76 @@
-                ProcessBuilder pb = new ProcessBuilder(cmd);
+        startDesktopSharing();
+        checkExpandingDesktopSharingLargeVideo(true);
+        testDesktopSharingInPresence("desktop");
 
-                pb.redirectOutput(ProcessBuilder.Redirect.INHERIT);
-                pb.redirectError(ProcessBuilder.Redirect.INHERIT);
-                Process p = pb.start();
-
-                waitStartSignal.countDown();
-
-                p.waitFor();
-                print("Script ended execution.");
-                waitEndSignal.countDown();
-            }
-            catch (IOException | InterruptedException e)
-            {
-                hookException = e;
-                waitStartSignal.countDown();
-            }
-        }).start();
-
-        final Participant owner = getParticipant1();
-
-        // now lets wait starting or error on startup
-        try
-        {
-            waitStartSignal.await(2, TimeUnit.SECONDS);
-            if (hookException != null)
-            {
-                hookException.printStackTrace();
-                fail("Error executing hook script:"
-                    + hookException.getMessage());
-            }
-        }
-        catch (InterruptedException e)
-        {
-            e.printStackTrace();
-        }
+    }
 
-        // let's wait some time the user to joins
-        TestUtils.waitForBoolean(
-            owner.getDriver(),
-            "return (APP.conference.membersCount == 2);",
-            25);
-        Participant ownerParticipant = getParticipant1();
-        ownerParticipant.waitForIceConnected();
-        ownerParticipant.waitForSendReceiveData();
-        ownerParticipant.waitForRemoteStreams(1);
+    /**
+     * Check desktop sharing stop.
+     */
+    @Test
+    public void testDesktopSharingStop()
+    {
+        stopDesktopSharing();
+        checkExpandingDesktopSharingLargeVideo(false);
+        testDesktopSharingInPresence("camera");
+    }
 
+    /**
+     * Checks the status of desktop sharing received from the presence and
+     * compares it to the passed expected result.
+     * @param expectedResult camera/desktop
+     */
+    private void testDesktopSharingInPresence(final String  expectedResult)
+    {
         // now lets check whether his stream is screen
-        String remoteParticipantID = owner.getDriver()
-            .findElement(By.xpath("//span[starts-with(@id, 'participant_') " +
-                " and contains(@class,'videocontainer')]")).getAttribute("id");
-        remoteParticipantID
-            = remoteParticipantID.replaceAll("participant_", "");
+        String participant1Jid
+            = MeetUtils.getResourceJid(getParticipant2().getDriver());
 
-        final String expectedResult = "desktop";
         // holds the last retrieved value for the remote type
         final Object[] remoteVideoType = new Object[1];
         try
         {
             final String scriptToExecute = "return APP.UI.getRemoteVideoType('"
```

Comment:
can we make this script execution a "getRemoteVideoType" method in WebParticipant ?

Reply:
Is there other place where we are using this? I don't think so. That's why I think this belongs to the DesktopSharingTest!

Reply:
No, it does not ! It belongs to the WebParticipant !

Reply:
Test class describes what is being tested and how. Who's video type is ? Participant's. In this particular case we're checking what video type given web participant has received for given remote participant through presence. Other tests can be also interested in checking that.